pkg/helper/autonode/helpers.go

```diff
@@ -0,0 +1,156 @@
+package autonode
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/interactive/confirm"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+// SetAutoNode validates and sets AutoNode configuration for a cluster
+func SetAutoNode(r *rosa.Runtime, cmd *cobra.Command, cluster *cmv1.Cluster, autonode string, roleArn string) (
+	autonodeMode string, autonodeRole string, err error) {
+
+	// Check if flags are changed
+	autonodeChanged := cmd.Flags().Changed("autonode")
+	roleArnChanged := cmd.Flags().Changed("autonode-iam-role-arn")
+
+	if !autonodeChanged && !roleArnChanged {
+		return "", "", nil
+	}
+
+	// Validate HCP cluster
+	if !aws.IsHostedCP(cluster) {
+		return "", "", fmt.Errorf("AutoNode is only supported for Hosted Control Plane clusters")
+	}
+
+	currentAutonodeEnabled := ocm.AutoNodeExists(cluster)
+
+	// Validate autonode value if provided
+	if autonodeChanged {
+		if autonode != "enabled" {
+			return "", "", fmt.Errorf("Invalid value for --autonode. Currently only 'enabled' is supported")
+		}
+
+		// Check if trying to enable when already enabled
+		if currentAutonodeEnabled {
+			return "", "", fmt.Errorf("AutoNode is already enabled for this cluster")
+		}
+
+		// When enabling AutoNode, IAM role ARN is required
+		if !roleArnChanged || roleArn == "" {
+			return "", "", fmt.Errorf("IAM role ARN is required when enabling AutoNode")
+		}
+
+		autonodeMode = autonode
```

Comment:
this should be in a validation function in the proper file IMO

Reply:
Agreed and did these in the last commit